packages/lib-webpack/src/webpack/GenerateManifestPlugin.ts

```diff
@@ -12,10 +17,28 @@ export class GenerateManifestPlugin implements WebpackPluginInstance {
           stage: Compilation.PROCESS_ASSETS_STAGE_ADDITIONAL,
         },
         () => {
+          const { entryChunk, runtimeChunk } = findPluginChunks(this.containerName, compilation);
+
+          const loadScripts = (runtimeChunk ? [runtimeChunk, entryChunk] : [entryChunk]).reduce<
+            string[]
+          >((acc, chunk) => [...acc, ...chunk.files], []);
+
+          const manifest: PluginManifest = {
+            ...this.manifestData,
+            loadScripts,
+            compilationHash: compilation.fullHash,
+          };
+
           compilation.emitAsset(
-            this.fileName,
-            new sources.RawSource(Buffer.from(JSON.stringify(this.manifest, null, 2))),
+            this.manifestFilename,
+            new sources.RawSource(Buffer.from(JSON.stringify(manifest, null, 2))),
           );
+
+          if (manifest.extensions.length === 0) {
+            const error = new WebpackError('Plugin manifest has no extensions');
```

Comment:
@vojtechszocs have we always asserted that a plugin must have 1 or more extensions? Why are we adding this stipulation now?

Reply:
This plugin SDK is built on the assumption that individual plugins will utilize extensions to extend host applications.

We didn't have this check before because ^^ assumption was implicit, now we're just making it more explicit. If a plugin doesn't provide any extensions (checked during plugin manifest generation), we emit a warning to the webpack compilation.

There may be existing plugins (like the ones for ConsoleDot) which communicate with their host application only via exposed modules. In order to support such plugins, we added new `PluginStore.getExposedModule` API which is basically an equivalent to [Scalprum `getCachedModule` function](https://github.com/scalprum/scaffloding/blob/main/packages/core/src/index.ts).